### PR TITLE
KNOX-2343 - Enhanced API services' display on the Knox Home page

### PR DESCRIPTION
**Pull request by smolnar82**
## What changes were proposed in this pull request?

I added `samples` into API-type `service.xml` files' `metadata` section (wherever applicable) and modified the underlying Java code to:
- interpret the new XML elements
- make samples available through the `metadata` API
- have them displayed on the Home page: instead of simply listing the Knox access of a particular service it now looks very similar to how we display the UI services. When clicking on the API service icon a new modal window gets opened with the selected API service details, including the newly added samples

## How was this patch tested?

Tested manually:
Screen Shot 2020-06-29 at 4 11 08 PM
Screen Shot 2020-06-29 at 4 11 33 PM
Screen Shot 2020-06-29 at 8 12 40 PM

### CODE REVIEW

File: gateway-service-definitions/src/main/resources/services/webhdfs/2.4.0/service.xml [hunk 1]
```diff
@@ -21,6 +21,27 @@
         <context>/webhdfs</context>
         <shortDesc>Web HDFS</shortDesc>
         <description>An HTTP REST API which supports the complete FileSystem interface for HDFS.</description>
+        <samples>
+            <sample>
+                <description>List all files under 'testPath'</description>
+                <method>GET</method>
+                <path>webhdfs/v1/testPath?op=LISTSTATUS</path>
```
Comment (smolnar82) [inline gateway-service-definitions/src/main/resources/services/webhdfs/2.4.0/service.xml:28]: 'webhdfs` should be removed!

**Comment by smolnar82** [comment]
Thanks, @moresandeep for your review comments, nice job spotting them out :) I fixed them all.

File: gateway-service-definitions/src/main/resources/services/cm-api/1.0.0/service.xml [hunk 2]
```diff
@@ -26,6 +26,22 @@
...
+                <path>clusters</path>
+            </sample>
+            <sample>
+                <description>Fetches HDFS service details from cluster named 'c1'</description>
+                <method>GET</method>
+                <path>clusters/c1/services/HDFS</path>
+            </sample>
+            <sample>
+                <description>You can checkout CM's API (v41) document here</description>
+                <value>https://archive.cloudera.com/cm7/7.1.1/generic/jar/cm_api/apidocs/</value>
```
Comment (risdenk) [inline gateway-service-definitions/src/main/resources/services/cm-api/1.0.0/service.xml:42]: I think you should link here instead? https://cloudera.github.io/cm_api/

This link will get out of date quickly?

File: gateway-service-definitions/src/main/resources/services/hbase/0.98.0/service.xml [hunk 3]
```diff
@@ -20,6 +20,32 @@
...
+                <path>namespaces/special_ns</path>
+            </sample>
+            <sample>
+                <description>Delete a namespace. The namespace must be empty.</description>
+                <method>DELETE</method>
+                <path>namespaces/special_ns</path>
+            </sample>
+            <sample>
+                <description>You may check out Apache Web HBase's REST API documentation here</description>
+                <value>https://docs.cloudera.com/documentation/enterprise/6/6.3/topics/admin_hbase_rest_api.html</value>
```
Comment (risdenk) [inline gateway-service-definitions/src/main/resources/services/hbase/0.98.0/service.xml:46]: This link would be better:

https://hbase.apache.org/book.html#_using_rest_endpoints

File: gateway-service-definitions/src/main/resources/services/webhdfs/2.4.0/service.xml [hunk 4]
```diff
@@ -21,6 +21,27 @@
...
+                <path>v1/testPath/testFile?op=RENAME&amp;destination=testPath/renamedFile</path>
+            </sample>
+            <sample>
+                <description>Get Home Directory</description>
+                <method>GET</method>
+                <path>v1/?op=GETHOMEDIRECTORY</path>
+            </sample>
+            <sample>
+                <description>You may check out Apache WebHDFS's REST API documentation here</description>
+                <value>https://hadoop.apache.org/docs/r1.0.4/webhdfs.html</value>
```
Comment (risdenk) [inline gateway-service-definitions/src/main/resources/services/webhdfs/2.4.0/service.xml:42]: This link is better:

https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-hdfs/WebHDFS.html

File: gateway-service-definitions/src/main/resources/services/zeppelinws/0.8.1/service.xml [hunk 5]
```diff
@@ -21,6 +21,27 @@
...
+                <path>api/notebook/job/note1</path>
+            </sample>
+            <sample>
+                <description>Delete a note by the given note id (e.g. 'note1')</description>
+                <method>DELETE</method>
+                <path>api/notebook/note1</path>
+            </sample>
+            <sample>
+                <description>You may check out Apache Zeppelin WS's REST API documentation here</description>
+                <value>https://zeppelin.apache.org/docs/0.7.0/rest-api/rest-notebook.html</value>
```
Comment (risdenk) [inline gateway-service-definitions/src/main/resources/services/zeppelinws/0.8.1/service.xml:42]: Should point to 0.8.1?

http://zeppelin.apache.org/docs/0.8.1/usage/rest_api/notebook.html

**Comment by smolnar82** [comment]
Thanks, @risdenk for your review comments. I fixed them all.

**Pull request merged**